File: sampleApp/src/test/java/com/nytimes/android/external/playbillingtester/sample/PrefsManagerTest.java
package com.nytimes.android.external.playbillingtester.sample;

import android.content.Context;

import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.MockitoAnnotations;
import org.robolectric.RobolectricTestRunner;
import org.robolectric.RuntimeEnvironment;
import org.robolectric.annotation.Config;

import static org.assertj.core.api.Assertions.assertThat;

@RunWith(RobolectricTestRunner.class)
@Config(manifest = Config.NONE)
public class PrefsManagerTest {

    private PrefsManager testObject;

    @Before
    public void setUp() throws Exception {
        MockitoAnnotations.initMocks(this);
        testObject = new PrefsManager(RuntimeEnvironment.application
                .getSharedPreferences("test", Context.MODE_PRIVATE));
    }

    @Test
    public void testSetAndGet() throws Exception {
        assertThat(testObject.isUsingTestGoogleServiceProvider()).isFalse();
        testObject.setUsingGoogleServiceProvider(true);
        assertThat(testObject.isUsingTestGoogleServiceProvider()).isTrue();
    }
}

File: sampleApp/src/test/java/com/nytimes/android/external/register/sample/PrefsManagerTest.kt
package com.nytimes.android.external.register.sample

import android.content.Context
import org.assertj.core.api.Assertions.assertThat
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith
import org.mockito.MockitoAnnotations
import org.robolectric.RobolectricTestRunner
import org.robolectric.RuntimeEnvironment
import org.robolectric.annotation.Config

@RunWith(RobolectricTestRunner::class)
@Config(manifest = Config.NONE)
class PrefsManagerTest {

    private var testObject: PrefsManager? = null

    @Before
    @Throws(Exception::class)
    fun setUp() {
        MockitoAnnotations.initMocks(this)
        testObject = PrefsManager(RuntimeEnvironment.application
                .getSharedPreferences("test", Context.MODE_PRIVATE))
    }

    @Test
    @Throws(Exception::class)
    fun testSetAndGet() {
        assertThat(testObject!!.isUsingTestGoogleServiceProvider).isFalse()
        testObject!!.setUsingGoogleServiceProvider(true)
        assertThat(testObject!!.isUsingTestGoogleServiceProvider).isTrue()
    }
}
